morphic.js

```diff
@@ -7818,10 +7820,47 @@ MenuMorph.prototype.select = function (aMenuItem) {
     this.selection = aMenuItem;
 };
 
+MenuMorph.prototype.mouseLeave = function () {
+    if (this.parent instanceof MenuMorph) {
+        // this is a submenu so we vanish when lose focus
+        var mouses = this.world.hand.allMorphsAtPointer();
+        if (contains(mouses, this.parent)) {
+            var items = mouses.filter(function (morph)
```

Comment:
All variables should be declared at the top of the function. Also the filter function should be 
```js
.filter(function () {
     return;
}
```

(it can also be extract into a named function `isMenuMorph` and then you'd do `.filter(isMenuMorph)` which looks much nicer.